include/s3select.h

```diff
@@ -2133,48 +2143,89 @@ class base_s3object
   }
 
   void result_values_to_string(multi_values& projections_resuls, std::string& result)
-  {
-    size_t i = 0;
+{   
+    std::string last_key;
     std::string output_delimiter(1,m_csv_defintion.output_column_delimiter);
     std::string output_row_delimiter(1,m_csv_defintion.output_row_delimiter);
+    std::vector<std::string> key_path(m_sa->get_json_key_path());
+    
+    if(key_path.size()) {
+      last_key = key_path[key_path.size()-1];
+    }
 
-    for(auto& res : projections_resuls.values)
-    {
-	    if(fp_ext_debug_mesg)
-		      fp_ext_debug_mesg( res->to_string() );
+    if(is_json)  {
```

Comment:
it's better to move the JSON result handling out of this method.
and if there are repetitive code segments, it is better to make these segments a method.